src/Chapter3.hs

```diff
@@ -374,6 +385,26 @@ after the fight. The battle has the following possible outcomes:
 
 -}
 
+data Knight = Knight {
+  knightName :: String,
+  knightHp :: Int,
+  knightAttack :: Int,
+  knightGold :: Int
+} deriving (Eq, Show)
+
+data Monster = Monster {
+  monsterHp :: Int,
+  monsterAttack :: Int,
+  monsterGold :: Int
+} deriving (Eq, Show)
+
+fight :: Knight -> Monster -> Int
+fight Knight{..} Monster{..}
```

Comment:
That's a very nice usage of RecordWildCards :tiger: 

Shameless plug, if you want to read more about best-practices around RecordWildCards, you can read my blog post on the subject :book: 

* https://kodimensional.dev/recordwildcards

Reply:
Will definitely take a look, thank you ! 😄